update_manifest.py

```diff
@@ -1,56 +1,135 @@
+        logging.critical(f"Manifest file in {base_path} has no element 'Version'")
+        return
+    if (new_version := version or old_version.get("number")) is None:
+        logging.critical(f"Manifest file in {base_path} has no attribute 'number'")
         return
-    root = manifest.getroot()
 
-    base_url = min((source.get("url") for source in root.iter("Source")), key=len)
-    parts = {
-        source.get("part"): pathlib.Path(source.get("url")).relative_to(base_url)
-        for source in root.iter("Source")
-    }
+    config = configparser.ConfigParser()
+    try:
+        config.read("manifest.cfg")
+    except FileNotFoundError:
+        logging.critical(f"Manifest configuration file not found in path '{base_path}'")
+        return
 
-    for file in root.iter("File"):
-        name, part = file.get("name"), file.get("part")
-        path = parts.get(part) / name
-        try:
+    base_url = "https://raw.githubusercontent.com/PathOfBuildingCommunity/PathOfBuilding/{branch}/"
+    parts: list[dict[str, str]] = []
+    for part in config.sections():
+        url = base_url + config[part]["path"]
+        url_with_trailing_slash = url if url.endswith("/") else url + "/"
+        attributes = {"part": part, "url": url_with_trailing_slash}
+        if part == "runtime":
+            attributes.update(platform="win32")
+        parts.append(attributes)
+
+    rules = {
+        "include-files": (_complement, _exclude_file),
+        "include-directories": (_complement, _exclude_directory),
+        "exclude-files": (_identity, _exclude_file),
+        "exclude-directories": (_identity, _exclude_directory),
+    }
+    files: list[dict[str, str]] = []
+    for section in config.sections():
+        exclusion_checks = [
+            modifier(functools.partial(check, set(config[section][option].split(","))))
+            for option, (modifier, check) in rules.items()
+            if config.has_option(section, option)
+        ]
+        source = pathlib.Path(config[section]["path"])
+        for path in source.glob("**/*.*"):
+            if any(is_excluded(path) for is_excluded in exclusion_checks):
+                continue
             data = path.read_bytes()
-        except FileNotFoundError:
-            logger.error(f"File not found, skipping {path}")
-            continue
-        sha1_hash = hashlib.sha1(data).hexdigest()
-        file.set("sha1", sha1_hash)
-        logger.info(f"Path: {path} hash: {sha1_hash}")
-    if version is not None:
-        root.find("Version").set("number", version)
-        logger.info(f"Updated to version {version}")
+            sha1 = hashlib.sha1(data).hexdigest()
+            name = path.relative_to(config[section]["path"]).as_posix()
+            attributes = {"part": section, "sha1": sha1, "name": name}
+            if path.suffix in [".dll", ".exe"]:
+                attributes.update(runtime="win32")
+            files.append(attributes)
```

Comment:
I like the look of the new manifest, as it lines everything up, but is there a way to order the attributes at least on this first run to verify the diff didn't change?  Right now everything looks like it changed, as the attributes changed order (before was name, part, sha1; now is part, sha1, name)